backend/cfg/cfg_liveness.ml

```diff
@@ -45,58 +45,48 @@ struct
       across : Reg.Set.t
     }
 
+  let instruction { before; across = _ } ~can_raise ~exn
+      (instr : _ Cfg.instruction) =
+    let across = Reg.diff_set_array before instr.res in
+    let across =
+      if can_raise then Reg.Set.union across exn.before else across
```

Comment:
We have lost the `instr.stack_offset > 0`
part of the condition.

Reply:
My believe that what is was checking was a heuristic that `instr.stack_offset = 0` implies `block.exn = None` because otherwise there would be something on stack.

Therefore removing is should yield the same results while being less complex and similarly fast.

Reply:
I agree the conditions are equivalent,
but I am still not sure why that part
of the condition can be removed.
Or more precisely, it is safe to remove
it but could result in a slightly bigger
live set.

Reply:
Ah, yes, I was confused between `exn`
the parameter and `exn` the field.